src/components/auction/AuctionDetails/index.tsx

```diff
@@ -210,20 +212,38 @@ const AuctionDetails = (props: Props) => {
         derivedAuctionInfo?.biddingToken,
         derivedAuctionInfo?.auctioningToken,
       )
-    const clearingPriceNumber = orderToPrice(clearingPriceInfoAsSellOrder)?.toSignificant(4)
+    const clearingPriceNumber =
+      showPriceInverted && orderToPrice(clearingPriceInfoAsSellOrder)
+        ? orderToPrice(clearingPriceInfoAsSellOrder).invert().toSignificant(5)
+        : orderToPrice(clearingPriceInfoAsSellOrder)?.toSignificant(5)
 
     return clearingPriceNumber ? (
-      <>
-        <TokenValue>{abbreviation(clearingPriceNumber)}</TokenValue>{' '}
-        <TokenSymbol>
-          {getTokenDisplay(derivedAuctionInfo?.biddingToken)} per{' '}
-          {getTokenDisplay(derivedAuctionInfo?.auctioningToken)}
-        </TokenSymbol>
-      </>
+      showPriceInverted ? (
+        <>
+          <TokenValue>{abbreviation(clearingPriceNumber)}</TokenValue>{' '}
+          <TokenSymbol>
+            {getTokenDisplay(derivedAuctionInfo?.auctioningToken)} per{' '}
+            {getTokenDisplay(derivedAuctionInfo?.biddingToken)}
+          </TokenSymbol>
+        </>
+      ) : (
+        <>
+          <TokenValue>{abbreviation(clearingPriceNumber)}</TokenValue>{' '}
+          <TokenSymbol>
+            {getTokenDisplay(derivedAuctionInfo?.biddingToken)} per{' '}
+            {getTokenDisplay(derivedAuctionInfo?.auctioningToken)}
+          </TokenSymbol>
+        </>
+      )
```

Comment:
I'm bad for names, but looking at this the only thing that changes is the order of the string inside `<TokenSymbol>` repeated twice. Maybe it could be extracted to a single value that depends of showPriceInverted, `useMemo` for example, or a function that receives `derivedAcutionInfo` and if they should be inverted or not.